### ISSUE TITLE
Plugin modiefies Backend JS Files

### ISSUE THREAD

**Issue by thlewandowski82** [FriendsOfShopware/FroshPlatformHtmlMinify#15]
### Question

Hello,
we have a Problem with the Plugin. We Use also Magnalister to list our Products on eBay and Amazon. This Plugin uses a iFrame and some JS from the local SW6 directory. Your Plugin modifies these Backend js so we got some trouble with magnalister.
Is it possible to suspend some directories from modification / compression?

Best regards

T. Lewandowski

**Comment by tinect** [comment]
Hi, Thank you for contacting us.

Sadly, I'd need additional information to check this and have no possibility to check magnalister on my end.

Could you please tell me which `route` it is?

Maybe we could add setting based on routes here https://github.com/FriendsOfShopware/FroshPlatformHtmlMinify/blob/4c0252ffc7b2eb936137b2523252bdf48c51a374/src/Listener/ResponseListener.php#L23

**Comment by thlewandowski82** [comment]
Hello, 
i dont know the exact route. 
How can i find it? 

This ist a Screenshot from the Backend. 


Mit freundlichen Grüßen 

Thomas Lewandowski 
Leiter E-Commerce 


Stellfeld & Ernst GmbH Flughafenstr. 151 44309 Dortmund 

Telefon: +49 (0) 231 - 54523-133 
Mobil: +49 (0) 173 - 3163542 
Telefax: +49 (0) 231 - 54523197 
E-Mail: [ ***@***.*** | ***@***.*** ] 



[ https://stellfeld-ernst.de/ | www.stellfeld-ernst.de ] 

Sitz der Gesellschaft und Rechnungsanschrift: 
Stellfeld & Ernst GmbH, Oberste-Wilms-Str. 22a, 44309 Dortmund, Deutschland 
Tel: +49 (0) 231 51981-0, Fax: +49 (0) 231 514004 
Geschäftsführung: Andreas Ernst | HRB 5318 AG Dortmund | USt-ID: DE124895100 


Von: "tinect" ***@***.***> 
An: "FriendsOfShopware/FroshPlatformHtmlMinify" ***@***.***> 
CC: "Thomas Lewandowski" ***@***.***>, "Author" ***@***.***> 
Gesendet: Montag, 8. Mai 2023 09:49:49 
Betreff: Re: [FriendsOfShopware/FroshPlatformHtmlMinify] Plugin modiefies Backend JS Files (Issue #15) 






Hi, Thank you for contacting us. 

Sadly, I'd need additional information to check this and have no possibility to check magnalister on my end. 

Could you please tell me which route it is? 

Maybe we could add setting based on routes here [ https://github.com/FriendsOfShopware/FroshPlatformHtmlMinify/blob/4c0252ffc7b2eb936137b2523252bdf48c51a374/src/Listener/ResponseListener.php#L23 | https://github.com/FriendsOfShopware/FroshPlatformHtmlMinify/blob/4c0252ffc7b2eb936137b2523252bdf48c51a374/src/Listener/ResponseListener.php#L23 ]

**Comment by tinect** [comment]
Hi,
you could ask a developer who can access your shop to get the route or ask magnalister.
Answering through mails seems not to add images to the github-issue. But I assume, this would not help.

But maybe I can download sourcecode from magnalister in the next days to check there. Please let me know any path you are accessing in the administration for this task.

**Comment by thlewandowski82** [comment]
Hello, 

i´ve got an answer from magnalister. Magnalister uses the following routes: 



    * @RouteScope(scopes={"storefront"}) @Route("/magnalister/{key}/{userId}", name="magnalister.admin.page", methods={"GET" , "POST"}, defaults={"XmlHttpRequest"=true}) 
    * @RouteScope(scopes={"storefront"}) @Route("/magnalister/", name="magnalister.front.page", methods={"GET" , "POST"}, defaults={"XmlHttpRequest"=true}) 


Mit freundlichen Grüßen 

Thomas Lewandowski 
Leiter E-Commerce 


Stellfeld & Ernst GmbH Flughafenstr. 151 44309 Dortmund 

Telefon: +49 (0) 231 - 54523-133 
Mobil: +49 (0) 173 - 3163542 
Telefax: +49 (0) 231 - 54523197 
E-Mail: [ ***@***.*** | ***@***.*** ] 



[ https://stellfeld-ernst.de/ | www.stellfeld-ernst.de ] 

Sitz der Gesellschaft und Rechnungsanschrift: 
Stellfeld & Ernst GmbH, Oberste-Wilms-Str. 22a, 44309 Dortmund, Deutschland 
Tel: +49 (0) 231 51981-0, Fax: +49 (0) 231 514004 
Geschäftsführung: Andreas Ernst | HRB 5318 AG Dortmund | USt-ID: DE124895100 


Von: "tinect" ***@***.***> 
An: "FriendsOfShopware" ***@***.***> 
CC: "Thomas Lewandowski" ***@***.***>, "Author" ***@***.***> 
Gesendet: Montag, 8. Mai 2023 09:58:47 
Betreff: Re: [FriendsOfShopware/FroshPlatformHtmlMinify] Plugin modiefies Backend JS Files (Issue #15) 





Hi, 
you could ask a developer who can access your shop to get the route or ask magnalister. 
Answering through mails seems not to add images to the github-issue. But I assume, this would not help. 

But maybe I can download sourcecode from magnalister in the next days to check there. Please let me know any path you are accessing in the administration for this task.

**Comment by thlewandowski82** [comment]
Hello, 
is it possible to exclude js files by name? 

Mit freundlichen Grüßen 

Thomas Lewandowski 
Leiter E-Commerce 


Stellfeld & Ernst GmbH Flughafenstr. 151 44309 Dortmund 

Telefon: +49 (0) 231 - 54523-133 
Mobil: +49 (0) 173 - 3163542 
Telefax: +49 (0) 231 - 54523197 
E-Mail: [ ***@***.*** | ***@***.*** ] 



[ https://stellfeld-ernst.de/ | www.stellfeld-ernst.de ] 

Sitz der Gesellschaft und Rechnungsanschrift: 
Stellfeld & Ernst GmbH, Oberste-Wilms-Str. 22a, 44309 Dortmund, Deutschland 
Tel: +49 (0) 231 51981-0, Fax: +49 (0) 231 514004 
Geschäftsführung: Andreas Ernst | HRB 5318 AG Dortmund | USt-ID: DE124895100 


Von: "tinect" ***@***.***> 
An: "FriendsOfShopware" ***@***.***> 
CC: "Thomas Lewandowski" ***@***.***>, "Author" ***@***.***> 
Gesendet: Montag, 8. Mai 2023 09:58:47 
Betreff: Re: [FriendsOfShopware/FroshPlatformHtmlMinify] Plugin modiefies Backend JS Files (Issue #15) 





Hi, 
you could ask a developer who can access your shop to get the route or ask magnalister. 
Answering through mails seems not to add images to the github-issue. But I assume, this would not help. 

But maybe I can download sourcecode from magnalister in the next days to check there. Please let me know any path you are accessing in the administration for this task.

**Comment by tinect** [comment]
This plugin does not change JS files. It changes/optimizes JS found inlined in a site.

It is currently not possible to exclude specific sites. But we can include the feature in the future, therefore I added tag „kind/enhancement“ onto this issue.

**Comment by tinect** [comment]
Sidenotes to magnalister:
Regarding magnalister added some fixes with the comment: 
`//we cannot use symfony response for magnalister, there is some cases that javascript code will minmized from Shopware core and that made a problem for javascript code of magnalister`. Besides this is wrong, it is bad they did not really resolve any issue - but just hacked anything.
Additionally, they are using storefront-routes for admin-tasks, this is really not recommended :-(

Basically, when a page is broken, this means for the most things: the JavaScript has issues or there are not known issue with the library we are using: https://github.com/mrclay/jsmin-php

I would actually know what the real problem is, but I don't have any magnalister access, I can just check the code, but not the backend of what they are collecting.

So, it would be nice, magnalister would check the issue and provide feedback. I'll try to contact them.

**Comment by tinect** [comment]
For the moment, I'll close here.

As already mentioned, Magnalister has `fixed` it on their own. Additionally, the support takes too much time to send any feedback. Therefore, finding a decent solution is not possible.

Currently, I don't see any reason to implement any function to exclude specified routes, while it is just because their JavaScript seems broken.
